xla/service/gpu/nccl_clique_key.h

```diff
@@ -59,8 +59,11 @@ constexpr static int64_t kAsyncStreamTotal =
 // Assigns a unique ID to a stream for asynchronous or synchronous execution.
 // These IDs can be used, for example, to look up the NCCL communicator.
 inline uint64_t GetStreamId(
-    bool is_async, AsyncStreamKind stream_kind = AsyncStreamKind::kCollective) {
-  return is_async ? static_cast<int64_t>(stream_kind) + 1 : 0;
+    uint64_t main_stream_id, bool is_async,
```

Comment:
This should not change, because this stream id != ExecutionStreamId, we need a better name, it's more a "kind of collective stream", but at run time it can be mapped to concrete ExecutionStreamId.

Can you add CollectiveStreamId type-safe int, so we can distinguish it from regular streams ids.